ui/decredmaterial/dropdown.go

```diff
@@ -21,127 +22,114 @@ type DropDown struct {
 	dropdownIcon   *widget.Icon
 	navigationIcon *widget.Icon
 	backdrop       *widget.Clickable
+	clickable      *Clickable
 
 	group            uint
 	closeAllDropdown func(group uint)
-	card             Card
 	Width            int
+	linearLayout     *LinearLayout
+	padding          layout.Inset
+	shadow           *Shadow
 }
 
 type DropDownItem struct {
-	Text   string
-	Icon   *Image
-	button *widget.Clickable
-	label  Label
+	Text      string
+	Icon      *Image
+	clickable *Clickable
+	label     Label
 }
 
 func (t *Theme) DropDown(items []DropDownItem, group uint) *DropDown {
-	c := &DropDown{
+	d := &DropDown{
 		theme:          t,
 		isOpen:         false,
-		items:          make([]DropDownItem, len(items)+1),
+		selectedIndex:  0,
+		items:          make([]DropDownItem, 0),
 		color:          t.Color.Gray1,
 		background:     t.Color.Surface,
 		dropdownIcon:   t.dropDownIcon,
 		navigationIcon: t.navigationCheckIcon,
 		backdrop:       new(widget.Clickable),
+		clickable:      t.NewClickable(true),
 
 		group:            group,
 		closeAllDropdown: t.closeAllDropdownMenus,
-		card:             t.Card(),
+		linearLayout: &LinearLayout{
+			Width:  WrapContent,
+			Height: WrapContent,
+			Border: Border{Radius: Radius(8)},
+		},
+		padding: layout.Inset{Top: values.MarginPadding8, Bottom: values.MarginPadding8},
+		shadow:  t.Shadow(),
 	}
 
+	d.clickable.HoverColor = Hovered(t.Color.InactiveGray)
+	d.clickable.Radius = Radius(8)
+
 	for i := range items {
-		items[i].button = new(widget.Clickable)
+		items[i].clickable = t.NewClickable(true)
 		items[i].label = t.Body1(items[i].Text)
-		c.items[i+1] = items[i]
-	}
-
-	if len(c.items) > 0 {
-		txt := items[0].Text
-		if len(items[0].Text) > 12 {
-			txt = items[0].Text[:12] + "..."
-		}
-
-		c.items[0] = DropDownItem{
-			Text:   items[0].Text,
-			Icon:   items[0].Icon,
-			label:  t.Body1(txt),
-			button: new(widget.Clickable),
-		}
-		c.selectedIndex = 1
+		d.items = append(d.items, items[i])
 	}
 
-	t.dropDownMenus = append(t.dropDownMenus, c)
-	return c
+	t.dropDownMenus = append(t.dropDownMenus, d)
+	return d
 }
 
-func (c *DropDown) Selected() string {
-	return c.items[c.SelectedIndex()+1].Text
+func (d *DropDown) Selected() string {
+	return d.items[d.SelectedIndex()].Text
 }
 
-func (c *DropDown) SelectedIndex() int {
-	return c.selectedIndex - 1
+func (d *DropDown) SelectedIndex() int {
+	return d.selectedIndex
 }
 
-func (c *DropDown) Len() int {
-	return len(c.items) - 1
+func (d *DropDown) Len() int {
+	return len(d.items)
 }
 
-func (c *DropDown) handleEvents() {
-	for c.items[0].button.Clicked() {
-		c.closeAllDropdown(c.group)
-		c.isOpen = !c.isOpen
-	}
-
-	for i := range c.items {
-		index := i
-		if index != 0 {
-			for c.items[index].button.Clicked() {
-				c.selectedIndex = index
-				txt := c.items[index].Text
-				if len(c.items[index].Text) > 12 {
-					txt = c.items[index].Text[:12] + "..."
-				}
-				c.items[0].label.Text = txt
```

Comment:
This lines of code help truncate the wallet name should incase the wallet name is too long. Currently, we dont have a limit to the length of the wallet name, and to prevent the wallet name dropdown from being too long, it was given a maximum width.

This is the effect of removing the code.
![image](https://user-images.githubusercontent.com/27733432/134657456-aa7ee058-46b3-465d-a4bf-b19a68312d1d.png)